tests/README.md

```diff
@@ -1,5 +1,79 @@
-# Python Tests against the arkouda_server
+# Python Tests against the arkouda\_server
 
-  * depending on your setup you may need to set `PYTHONPATH` environment variable to find the `arkouda.py` module
-  * check the python source for runtime options
-  
\ No newline at end of file
+All Python tests can be run against a local or remote arkouda\_server. The arkouda project is configured so that
+no special Python paths need to be defined.
+  
+# The arkouda Python test harness
+
+The arkouda project has a [pytest](https://docs.pytest.org/en/latest/)-based Python test harness that can be executed 
+via the command-line interface (cli) or within the make file. 
+
+## pytest environment preparation
+
+There are two python libraries that must be installed to execute the arkouda test harness, pytest and pytest-env,
+both of which can be installed via pip:
+```
+pip install pytest, pytest-env
+```
+## Configuration of pytest: the pytest.ini file
+
+The pytest.ini file configures the arkouda Python test harness, specifically the following parameters:
+
+```
+[pytest]
+testpaths = 
+    tests/base_test.py
+    tests/client_test.py
+    tests/compare_test.py
+    tests/dtypes_tests.py
+    tests/groupby_test.py
+    tests/join_test.py
+    tests/operator_tests.py
+    tests/setops_test.py
+    tests/string_test.py
+    tests/where_test.py
+norecursedirs = .git dist build *egg* tests/deprecated/*
+python_functions = test*
+env =
+    ARKOUDA_SERVER_HOST=localhost
+    ARKOUDA_SERVER_PORT=5555
+    ARKOUDA_FULL_STACK_TEST=False
+    ARKOUDA_VERBOSE=False
+```
+* testpaths: shows the paths to all test files. For the time-being, the arkouda unit tests to be executed are 
+specified on a per-file basis, but pytest can also support directory-level configuration. 
+* norecursedirs: indicates which directories contain files that should be ignored, including tests/deprecated
+* python\_functions: the naming pattern for all test functions. Ih the case of arkouda, all python method (functions)
+  with names starting with "test" are executed by pytest
+* env: the pytest env variables needed to execute the arkouda Python test harness 
+
+## arkouda pytest environmental variables
+* ARKOUDA\_SERVER\_HOST: the hostname or ip address where the arkouda\_server is located
+* ARKOUDA\_SERVER\_PORT: the port the arkouda\_server is listening on
+* ARKOUDA\_FULL\_STACK_TEST: if True, the TestCase.setUpClass method starts up an arkouda\_server on the local machine, where
+  server=ARKOUDA\_SERVER\_HOST and port= ARKOUDA\_SERVER\_PORT. If False, the test harness runs in client mode and 
+  consequently an arkouda\_server is not started up.
+
+
+# Running the arkouda Python test harness
+
+To execute all tests in the arkouda test harness via the command line, execute the following command:
+
+```
+pytest -c pytest.ini 
```

Comment:
If you're on board with the `python3 -m pytest` suggestion, probably change these docs too.

Reply:
Agreed, will do